src/tools/kata-ctl/src/ops/env_ops.rs

```diff
@@ -0,0 +1,468 @@
+            .ok_or("could not find hypervisor config in configuration")
+            .map_err(|e| anyhow!(e))?;
+    }
+
+    let version =
+        get_command_version(&hypervisor_config.path).context("error getting hypervisor version")?;
+
+    let hypervisor_info = HypervisorInfo {
+        machine_type: hypervisor_config.machine_info.machine_type.to_string(),
+        machine_accelerators: hypervisor_config
+            .machine_info
+            .machine_accelerators
+            .to_string(),
+        version,
+        path: hypervisor_config.path.to_string(),
+        block_device_driver: hypervisor_config
+            .blockdev_info
+            .block_device_driver
+            .to_string(),
+        entropy_source: hypervisor_config.machine_info.entropy_source.to_string(),
+        shared_fs: hypervisor_config
+            .shared_fs
+            .shared_fs
+            .clone()
+            .unwrap_or(String::from("none")),
+        virtio_fs_daemon: hypervisor_config.shared_fs.virtio_fs_daemon.to_string(),
+        msize_9p: hypervisor_config.shared_fs.msize_9p,
+        memory_slots: hypervisor_config.memory_info.memory_slots,
+        pcie_root_port: hypervisor_config.device_info.pcie_root_port,
+        hotplug_vfio_on_rootbus: hypervisor_config.device_info.hotplug_vfio_on_root_bus,
+        debug: hypervisor_config.debug_info.enable_debug,
+        enable_iommu: hypervisor_config.device_info.enable_iommu,
+        enable_iommu_platform: hypervisor_config.device_info.enable_iommu_platform,
+        default_vcpus: hypervisor_config.cpu_info.default_vcpus,
+        cpu_features: hypervisor_config.cpu_info.cpu_features.to_string(),
+    };
+
+    let image_info = ImageInfo {
+        path: hypervisor_config.boot_info.image.clone(),
+    };
+
+    let kernel_info = KernelInfo {
+        path: hypervisor_config.boot_info.kernel.to_string(),
+        parameters: hypervisor_config.boot_info.kernel_params.to_string(),
+    };
+
+    let initrd_info = InitrdInfo {
+        path: hypervisor_config.boot_info.initrd.to_string(),
+    };
+
+    Ok((hypervisor_info, image_info, kernel_info, initrd_info))
+}
+
+pub fn get_env_info(toml_config: &TomlConfig) -> Result<EnvInfo> {
+    let metainfo = get_meta_info();
+
+    let runtime_info = get_runtime_info(toml_config);
+
+    let agent_info = get_agent_info(toml_config)?;
+
+    let host_info = get_host_info()?;
+
+    let (hypervisor_info, _image_info, kernel_info, initrd_info) =
+        get_hypervisor_info(toml_config)?;
+
+    let env_info = EnvInfo {
+        meta: metainfo,
+        runtime: runtime_info,
+        kernel: kernel_info,
+        image: _image_info,
+        initrd: initrd_info,
+        hypervisor: hypervisor_info,
+        host: host_info,
+        agent: agent_info,
+    };
+
+    Ok(env_info)
+}
+
+pub fn handle_env(env_args: EnvArgument) -> Result<()> {
+    let (toml_config, _) = TomlConfig::load_raw_from_file("").context("load toml config")?;
```

Comment:
This looks a bit odd as there is no file specified. Is there an alternative (`TomlConfig::Default()` or similiar?)

Reply:
@jodh-intel The current API does not have that, if the config_file provided to load_raw_from_file(config_file) is empty, it is only then that the kata-types crate searches for config file in the order configured.

On a side note, I was thinking it will be useful to support passing a configuration file path to the kata-ctl env, that way the tool can show environment wrt to the config file provided. Wdyt?

Reply:
Sounds good, but feel free to raise an issue and add that on a separate PR if you want.